pytket/pytket/passes/auto_rebase.py

```diff
@@ -44,6 +44,13 @@ def _TK2_using_TK2(a: Param, b: Param, c: Param) -> Circuit:
     OpType.ZZMax: _library._TK2_using_ZZMax,
 }
 
+_TK2_CIRCS_WIRE_SWAP: Dict[OpType, Callable[[Param, Param, Param], "Circuit"]] = {
+    OpType.TK2: _TK2_using_TK2,
+    OpType.ZZPhase: _library._TK2_using_ZZPhase,
```

Comment:
Shouldn't we add a special version of this one too?

Reply:
For the TK2 case, it may be best to just check if the angles are equivalent to a SWAP, in which case adding a wire SWAP, else completing like usual. There's no advantage to iSWAPMax or Sycamore gates having a wire swap as they would still have the same TK2 cost.

For ZZPhase, we could do the ZZMax conversion and then convert to ZZPhase. Currently we always convert TK2 to 3 ZZPhase + local Clifford corrections. This logic doesn't fit cleanly with the function added in `CircPool`, but I could refactor it to isolate a check for the the number of two qubit gates predicted. If this is < 3 then we could do the construct as CX and convert to ZZPhase, else we could do the normal version.

Reply:
+1 to both suggestions.

Reply:
This is now done